deps/eventlog/src/ccel/boot_services_app.rs

```diff
@@ -80,33 +80,20 @@ fn get_nested_paths(device_path_bytes: &[u8], mut result: EventDetails) -> Resul
     get_nested_paths(device_path, result)
 }
 
-/// Initial approach for parsing text device nodes described in specification:
-/// <https://uefi.org/specs/UEFI/2.10/10_Protocols_Device_Path_Protocol.html#text-device-node-reference>
-/// TODO Implement full support for data paths available for device nodes
+/// Prints device path based on type and subtype.
+/// For unsupported paths default formatter will be used: Path(type,subtype,hex_data).
 fn print_path(efi_type: u8, efi_sub_type: u8, vendor_data: &[u8]) -> String {
-    // Type: 1 (Hardware Device Path) / SubType: 1 (PCI)
-    if efi_type == 1 && efi_sub_type == 1 && vendor_data.len() == 2 {
-        let func_num = vendor_data[0];
-        let device_num = vendor_data[1];
-        return format!("Pci({},{})", func_num, device_num);
-    }
-
-    // Type: 4 (Media Device Path) / SubType: 4 (File Path)
-    if efi_type == 4 && efi_sub_type == 4 && (vendor_data.len() % 2 == 0) {
-        let utf16_words: Vec<u16> = vendor_data[0..vendor_data.len() - 2]
-            .chunks_exact(2)
-            .map(LittleEndian::read_u16)
-            .collect();
-        return format!("File({})", String::from_utf16_lossy(&utf16_words));
+    match DevicePathDispatcher::parse(efi_type, efi_sub_type, vendor_data) {
+        Ok(text) => text,
+        Err(_) => {
+            format!(
+                "Path({},{},{})",
+                efi_type,
+                efi_sub_type,
+                hex::encode(vendor_data)
+            )
+        }
```

Comment:
match patterns that extract w/ a default fallback can be often written more succinctly:

```suggestion
    DevicePathDispatcher::parse(efi_type, efi_sub_type, vendor_data)
      .unwrap_or_else(|| format!("Path({},{},{})", efi_type, efi_sub_type, hex::encode(vendor_data)))
```

Reply:
Applied and adjusted to new approach